libyara/modules/cuckoo.c

```diff
@@ -136,52 +136,201 @@ define_function(network_http_post)
 }
 
 
+#define REGISTRY_KEY_ACCESS         0x00
+#define REGISTRY_KEY_READ           0x01
+#define REGISTRY_KEY_WRITE          0x02
+#define REGISTRY_KEY_DELETE         0x03
+#define REGISTRY_KEY_VALUE_ACCESS   0x04
+
+
+uint64_t registry_match(
+  YR_OBJECT* registry_obj,
+  RE_CODE name_regexp,
+  int registry_operation_type)
+{
+  static const char* const registry_names[] = {
+    "keys",
+    "read_keys",
+    "write_keys",
+    "delete_keys"
+  };
+
+  json_t* summary_json = json_object_get((json_t*) registry_obj->data, "summary");
+  json_t* registry_json = json_object_get(summary_json, registry_names[registry_operation_type]);
+  json_t* value;
+
+  uint64_t result = 0;
+  size_t index;
+
+  json_array_foreach(registry_json, index, value)
+  {
+    if (yr_re_match(name_regexp, json_string_value(value)) > 0)
+    {
+      result = 1;
+      break;
+    }
+  }
+
+  return result;
+}
+
+
 define_function(registry_key_access)
 {
+  return_integer(
+    registry_match(
+      parent(),
+      regexp_argument(1),
+      REGISTRY_KEY_ACCESS));
+}
+
+
+define_function(registry_key_read)
+{
+  return_integer(
+    registry_match(
+      parent(),
+      regexp_argument(1),
+      REGISTRY_KEY_READ));
+}
+
+
+define_function(registry_key_write)
+{
+  return_integer(
+    registry_match(
+      parent(),
+      regexp_argument(1),
+      REGISTRY_KEY_WRITE));
+}
+
+
+define_function(registry_key_delete)
+{
+  return_integer(
+    registry_match(
+      parent(),
+      regexp_argument(1),
+      REGISTRY_KEY_DELETE));
+}
+
+
+define_function(registry_key_value_access)
+{
   YR_OBJECT* registry_obj = parent();
 
-  json_t* keys_json = (json_t*) registry_obj->data;
+  json_t* enhanced_json = json_object_get((json_t*) registry_obj->data, "enhanced");
```

Comment:
Uh, I'm sorry, I realised my changes are for this fork: https://github.com/spender-sandbox/cuckoo-modified , and not for vanilla cuckoo. Therefore feel free to ignore this pull request. However I made similar changes which support vanilla cuckoo 2 and I'm planning to submit the pull request soon.